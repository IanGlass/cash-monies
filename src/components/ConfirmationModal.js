import React from 'react';
import Modal from 'react-modal';

const ConfirmationModal = (props) => (
  <Modal
    isOpen={props.showModal}
    onRequestClose={props.closeModal}
    ariaHideApp={false}
    className="modal"
    contentLabel="Confirm Action"
  >
    <p>Are you sure?</p>
    <div className="button--group">
      <button className="button" onClick={props.confirmAction}>Yes</button>
      <button className="button" onClick={props.closeModal}>No</button>
    </div>
  </Modal>
);

export default ConfirmationModal;