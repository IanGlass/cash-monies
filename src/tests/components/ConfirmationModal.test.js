import React from 'react';
import { shallow } from 'enzyme';

import ConfirmationModal from '../../components/ConfirmationModal';

let wrapper, closeModal, confirmAction

beforeEach(() => {
  closeModal = jest.fn();
  confirmAction = jest.fn();
  wrapper = shallow(<ConfirmationModal closeModal={closeModal} confirmAction={confirmAction} />)
});

test('should render ConfirmationModal', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should execute confirmAction in ConfirmationModal', () => {
  wrapper.find('button').first().simulate('click');
  expect(confirmAction).toHaveBeenCalled();
})

test('should close ConfirmationModal', () => {
  wrapper.find('button').last().simulate('click');
  expect(closeModal).toHaveBeenCalled();
});