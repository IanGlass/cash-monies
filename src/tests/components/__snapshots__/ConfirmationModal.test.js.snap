// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConfirmationModal 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={0}
  contentLabel="Confirm Action"
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
>
  <p>
    Are you sure?
  </p>
  <div
    className="button--group"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      Yes
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      No
    </button>
  </div>
</Modal>
`;
